internal/testing/go.mod

```diff
@@ -0,0 +1,87 @@
+module github.com/pressly/goose/v3/internal/testing
+
+go 1.22.1
```

Comment:
This enables us to keep up-to-date with drivers without having to fight them on the minimum supported Go version.

This will especially be useful if we go down this route with https://github.com/pressly/goose/issues/664